modeler.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABCMeta, abstractmethod
+
+class Dynamics(metaclass=ABCMeta):
+    """Interface to dynamics model."""
+
+    @abstractmethod
+    def __call__(self, state, action):
+        """Simulate a world transition.
+
+        Args:
+          state (object): Current world state to start from.
+          action (object): Action to take in current world state.
+        
+        Returns:
+          (object): Next state after taking the given action in the given state.
+        """
+        pass
+
+class PerfectDynamics(Dynamics, metaclass=ABCMeta):
+    """Interface to perfect information dynamics model."""
+
+    @abstractmethod
+    def __call__(self, state, action):
+        """Simulate a world transition.
+
+        Args:
+          state (object): Current world state to start from.
+          action (object): Action to take in current world state.
+        
+        Returns:
+          (object): Next state.
+          (float): Reward or final state value.
+          (bool): Is terminal.
+        """
+        pass
+
+
+class Modeler(metaclass=ABCMeta):
```

Comment:
typo? Modeler vs Modeller

Reply:
Nope: https://www.diki.pl/slownik-angielskiego?q=modeler

Reply:
Buy I agree it could have different name. Any ideas?

Reply:
EnvExpert, EnvEstimator, EnvSimulator, Scout

Reply:
Hmm Scout brzmi nieźle, ale jak to uargumentujemy xD Że przeszukuje możliwe stany i uczy model? 😛